File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/gn_gdo_string_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* GNSDK internal gdo string provider class
*/
public class gn_gdo_string_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal gn_gdo_string_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(gn_gdo_string_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~gn_gdo_string_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_gn_gdo_string_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public gn_gdo_string_provider(GnDataObject obj, string key) : this(gnsdk_csharp_marshalPINVOKE.new_gn_gdo_string_provider(GnDataObject.getCPtr(obj), key), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public string get_data(uint pos) {
    string ret = gnsdk_csharp_marshalPINVOKE.gn_gdo_string_provider_get_data(swigCPtr, pos);
    return ret;
  }

  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.gn_gdo_string_provider_count(swigCPtr);
    return ret;
  }

  public static readonly uint kOrdinalStart = gnsdk_csharp_marshalPINVOKE.gn_gdo_string_provider_kOrdinalStart_get();
  public static readonly uint kCountOffset = gnsdk_csharp_marshalPINVOKE.gn_gdo_string_provider_kCountOffset_get();
}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/gn_gdo_string_provider.java

package com.gracenote.gnsdk;

/** 
* GNSDK internal gdo string provider class 
*/ 
 
public class gn_gdo_string_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected gn_gdo_string_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(gn_gdo_string_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_gn_gdo_string_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public gn_gdo_string_provider(GnDataObject obj, String key) {
    this(gnsdk_javaJNI.new_gn_gdo_string_provider(GnDataObject.getCPtr(obj), obj, key), true);
  }

  public String getData(long pos) {
    return gnsdk_javaJNI.gn_gdo_string_provider_getData(swigCPtr, this, pos);
  }

  public long count() {
    return gnsdk_javaJNI.gn_gdo_string_provider_count(swigCPtr, this);
  }

}
